phoenix-core/src/it/java/org/apache/phoenix/end2end/PermissionNSEnabledIT.java

```diff
@@ -50,4 +60,43 @@ public Void run() throws Exception {
             revokeAll();
         }
     }
+
+    @Test
+    public void testConnectionCreationFailsWhenNoExecPermsOnSystemCatalog() throws Throwable {
+        try {
+            grantSystemTableAccess();
+            superUser1.runAs((PrivilegedExceptionAction<Object>) () -> {
```

Comment:
Patch looks fine for master, but isn't the "->" lambda operator a Java 8 thing? If so you'll have to modify when backporting to 4.x, because from prior discussions on the dev list the community decided to keep 4.x compatible with JDK 7 because HBase 1.x is.

Reply:
Or alternatively do all versions in a JDK 7 compatible way

Reply:
@gjacoby126 yup, I realized this when I tried applying it to 4.x branches and so I put up a different patch for those branches. Thanks!